submissions/nataliaArtsiukh/Frogger-game/app.js

```diff
@@ -0,0 +1,168 @@
+const sizes = {
+    colWidth: 101,
+    rowHeight: 83,
+    enemyVerticalShift: 20,
+    playerVerticalShift: 10,
+    startRow: 5,
+    startCol: 2,
+    starVerticalShift: 10,
+    speedCoefficient: 0.7,
+    enemyHorizontalShift: 20,
+    colCount: 5,
+    rowCount: 6,
+    starCount: 5,
+    enemyCount: 3
+}
+
+const Star = function(numCol) {
+    this.x = sizes.colWidth * numCol; 
+    this.y = -sizes.starVerticalShift;
+    this.sprite = 'images/Star.png';
+}
+
+Star.prototype.update = function() {
+    if (player.y < 0) {
+        let index = player.x / sizes.colWidth; 
```

Comment:
Here the code implies global `player` variable exists.
Classes shouldn't know anything outer world unless they get this information explicitly (e.g. a reference to a foreign object is passed to them at construction time and stored among class properties for re-use).

Direct reference to global environmental variables (like `Resources`) is OK here. This is a flaw of the game engine architecture, but we do not want to fix it. This happens on real-world projects.

Reply:
The comment above remains valid. Key part is "at construction".
Engine's `update` function has its own signature. It doesn't pass any variables to game characters and we do not want to interfere with it.
![image](https://user-images.githubusercontent.com/8686631/185862115-aeeeb9a5-7afc-4a80-8081-57215ea394cd.png)

The only way to resolve is to pass the `player` at object instantiation time, store it in enemies/stars instances properties just like coordinates or whatever we store locally and re-use it (local reference to the player) in update/collision testing method.

Reply:
I made the necessary changes. I can't click re-review. please check)